py_ecc/fields/field_elements.py

```diff
@@ -0,0 +1,377 @@
+            on = other
+        else:
+            raise TypeError(
+                "Expected an int or FQ object, but got object of type {}"
+                .format(type(other))
+            )
+
+        return type(self)((on - self.n) % self.field_modulus)
+
+    def __sub__(self, other: IntOrFQ) -> "FQ":
+        if isinstance(other, FQ):
+            on = other.n
+        elif isinstance(other, int):
+            on = other
+        else:
+            raise TypeError(
+                "Expected an int or FQ object, but got object of type {}"
+                .format(type(other))
+            )
+
+        return type(self)((self.n - on) % self.field_modulus)
+
+    def __div__(self, other: IntOrFQ) -> "FQ":
+        if isinstance(other, FQ):
+            on = other.n
+        elif isinstance(other, int):
+            on = other
+        else:
+            raise TypeError(
+                "Expected an int or FQ object, but got object of type {}"
+                .format(type(other))
+            )
+
+        return type(self)(
+            self.n * prime_field_inv(on, self.field_modulus) % self.field_modulus
+        )
+
+    def __truediv__(self, other: IntOrFQ) -> "FQ":
+        return self.__div__(other)
+
+    def __rdiv__(self, other: IntOrFQ) -> "FQ":
+        if isinstance(other, FQ):
+            on = other.n
+        elif isinstance(other, int):
+            on = other
+        else:
+            raise TypeError(
+                "Expected an int or FQ object, but got object of type {}"
+                .format(type(other))
+            )
+
+        return type(self)(
+            prime_field_inv(self.n, self.field_modulus) * on % self.field_modulus
+        )
+
+    def __rtruediv__(self, other: IntOrFQ) -> "FQ":
+        return self.__rdiv__(other)
+
+    def __pow__(self, other: int) -> "FQ":
+        if other == 0:
+            return type(self)(1)
+        elif other == 1:
+            return type(self)(self.n)
+        elif other % 2 == 0:
+            return (self * self) ** (other // 2)
+        else:
+            return ((self * self) ** int(other // 2)) * self
+
+    def __eq__(self, other: IntOrFQ) -> bool:  # type:ignore # https://github.com/python/mypy/issues/2783 # noqa: E501
+        if isinstance(other, FQ):
+            return self.n == other.n
+        elif isinstance(other, int):
+            return self.n == other
+        else:
+            raise TypeError(
+                "Expected an int or FQ object, but got object of type {}"
+                .format(type(other))
+            )
+
+    def __ne__(self, other: IntOrFQ) -> bool:    # type:ignore # https://github.com/python/mypy/issues/2783 # noqa: E501
+        return not self == other
+
+    def __neg__(self) -> "FQ":
+        return type(self)(-self.n)
+
+    def __repr__(self) -> str:
+        return repr(self.n)
+
+    def __int__(self) -> int:
+        return self.n
+
+    @classmethod
+    def one(cls) -> "FQ":
+        return cls(1)
```

Comment:
Not for this PR but a possible performance gain might be available if we interned (cached) these return values so that we don't need to re-instantiate the `1` and `0` values each time.

Reply:
Opened #55 for this